services/backend/src/core/geds/geds.js

```diff
@@ -43,54 +43,284 @@ async function getGedsAssist(request, response) {
+
+      const dataGEDS = dataGEDSArray.find((element) => {
+        return element.contactInformation.email === dataDBEmail;
+      });
+
+      const organizations = [];
+      let organizationCounter = 0;
+      for (
+        let currentBranch = dataGEDS;
+        currentBranch.organizationInformation;
+        currentBranch = currentBranch.organizationInformation.organization
+      ) {
+        const branchInfo = {
+          description:
+            currentBranch.organizationInformation.organization.description,
+          tier: organizationCounter,
+          addressInformation:
+            currentBranch.organizationInformation.organization
+              .addressInformation,
+          id: currentBranch.organizationInformation.organization.id,
         };
+        organizationCounter += 1;
+        organizations.unshift(branchInfo);
+      }
+
+      const branchOrg = organizations[Math.min(2, organizations.length - 1)];
+
+      const enAddr = branchOrg.addressInformation.address.en;
+
+      try {
+        const location = await prisma.opOfficeLocation.findMany({
+          where: {
+            country: branchOrg.addressInformation.country.en,
+            city: branchOrg.addressInformation.city.en,
+            postalCode: branchOrg.addressInformation.pc,
+            streetNumber: parseInt(enAddr.split(" ")[0], 10),
+          },
+        });
+
+        const profile = {};
+
+        if (dataGEDS.givenName && dataGEDS.givenName !== user.firstName) {
+          profile.firstName = dataGEDS.givenName;
+        }
+
+        if (dataGEDS.surname && dataGEDS.surname !== user.lastName) {
+          profile.lastName = dataGEDS.surname;
+        }
+
+        if (dataGEDS.telephone && dataGEDS.telephone !== user.telephone) {
+          profile.telephone = dataGEDS.telephone;
+        }
+
+        if (dataGEDS.cellphone && dataGEDS.cellphone !== user.cellphone) {
+          profile.cellphone = dataGeds.cellphone;
+        }
+
+        if (!user.officeLocation || location[0].id !== user.officeLocation.id) {
+          profile.locationId = location[0].id;
+        }
+
+        if (dataGEDS.title || dataGEDS.branch) {
+          let updateEmployeeInfo = !(
+            user.employmentInfo && user.employmentInfo.translations
+          );
+          if (!updateEmployeeInfo) {
+            user.employmentInfo.translations.forEach((translation) => {
+              if (
+                translation.jobTitle !==
+                  dataGEDS.title[
+                    translation.language === "ENGLISH" ? "en" : "fr"
+                  ] ||
+                translation.branch !==
+                  branchOrg.description[
+                    translation.language === "ENGLISH" ? "en" : "fr"
+                  ]
+              ) {
+                updateEmployeeInfo = true;
+              }
+            });
+          }
+
+          if (updateEmployeeInfo) {
+            profile.jobTitle = {
+              ENGLISH: dataGEDS.title.en,
```

Comment:
Since we are only returning one language at a time I am not sure if this will work. What do you think @BenJeau

Reply:
I'm not too sure I understand. The geds functions return both languages for values and did before the changes of this PR too.

Reply:
Oh, I was just wondering if the backend was sending both languages to the frontend. Since with the new API, we only return one language. You can ignore this comment. If it works then its good.